test/test_roscpp/test/src/params.cpp

```diff
@@ -528,6 +528,38 @@ TEST(Params, mapBoolParam)
   ASSERT_TRUE(std::equal(map_b.begin(), map_b.end(), map_b2.begin()));
 }
 
+TEST(Params, paramTemplateFunction)
+{
+  EXPECT_EQ( param::param<std::string>( "string", "" ), "test" );
+  EXPECT_EQ( param::param<std::string>( "gnirts", "test" ), "test" );
+
+  EXPECT_TRUE( param::param<int>( "int", 0 ) == 10 );
+  EXPECT_TRUE( param::param<int>( "tni", 10 ) == 10 );
+
+  EXPECT_DOUBLE_EQ( param::param<double>( "double", 0.0 ), 10.5 );
+  EXPECT_DOUBLE_EQ( param::param<double>( "elbuod", 10.5 ), 10.5 );
+
+  EXPECT_FALSE( param::param<bool>( "bool", true ) );
+  EXPECT_TRUE( param::param<bool>( "loob", true ) );
+}
+
+TEST(Params, paramNodeHandleTemplateFunction)
+{
+  NodeHandle nh;
+
+  EXPECT_EQ( nh.param<std::string>( "string", "" ), "test" );
+  EXPECT_EQ( nh.param<std::string>( "gnirts", "test" ), "test" );
+
+  EXPECT_TRUE( nh.param<int>( "int", 0 ) == 10 );
+  EXPECT_TRUE( nh.param<int>( "tni", 10 ) == 10 );
+
+  EXPECT_DOUBLE_EQ( nh.param<double>( "double", 0.0 ), 10.5 );
+  EXPECT_DOUBLE_EQ( nh.param<double>( "elbuod", 10.5 ), 10.5 );
+
+  EXPECT_FALSE( nh.param<bool>( "bool", true ) );
```

Comment:
In this case I think it doesn't matter. The most important thing is to use `EXPECT_EQ` on non-bool types so that when they do not match it can print out what the actual versus expected was. In this case there are only to states, so being explicit doesn't help.